flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/AnalyzeTableUtil.java

```diff
@@ -97,14 +102,28 @@ public static TableResultInternal analyzeTable(
                     executeSqlAndGenerateStatistics(tableEnv, columns, statSql);
             CatalogTableStatistics tableStat = result.f0;
             catalog.alterTableStatistics(objectPath, tableStat, false);
-            CatalogColumnStatistics columnStat = result.f1;
-            if (columnStat != null) {
-                catalog.alterTableColumnStatistics(objectPath, columnStat, false);
+            CatalogColumnStatistics newColumnStat = result.f1;
+            if (newColumnStat != null) {
+                CatalogColumnStatistics oldColumnStat =
+                        catalog.getTableColumnStatistics(objectPath);
+                // merge stats.
+                CatalogColumnStatistics mergedColumnStatistics =
+                        mergeColumnStatistics(oldColumnStat.copy(), newColumnStat);
```

Comment:
It's better we can move the `copy` operation into `mergeColumnStatistics ` method

Reply:
> It's better we can move the `copy` operation into `mergeColumnStatistics ` method

Done!